### ISSUE TITLE
Upgrade Rocket.Chat through Admin UI?

### ISSUE THREAD

**Issue by brylie** [RocketChat/Rocket.Chat#2618]
Your Rocket.Chat version: 0.19

```
As a system administrator
I would like to upgrade Rocket.Chat through the Admin UI
so that our team can Rocket.Chat keep up-to-date with minimal effort
```
# Feature

Add a page to the Administrator UI allowing privileged users to upgrade Rocket.Chat to the latest version. This feature has been demonstrated by applications such as Discourse (which also uses Docker for deployment).

What are the plans to support Rocket.Chat upgrades (and possibly backups) via the Admin UI?
# Background

We have multiple people with administrator privileges in the Rocket.Chat application, so our team could easily manage Rocket.Chat upgrades through the UI. 

However, we have fewer people who have access to the Rocket.Chat deployment server, and who are also familiar with Docker to perform the upgrade. This means we are continually finding ourselves lagging behind the latest version of Rocket.Chat by several version numbers.

**Comment by andreas-bulling** [comment]
+1

**Comment by k0nsl** [comment]
I like this suggestion, particularly your suggestion about being able to backup the instance from within the Admin UI.

However, as for upgrading: is there some way to distinguish if Rocket.Chat has been deployed with docker, or if it has been deployed simply by launching `meteor`? I don't use docker for Rocket.Chat myself, hence the curiosity.

**Comment by IQ2022** [comment]
:+1: :100: backup DB, or even email list, etc (ability to download a copy) for future deployment. 
Also the version upgrade is great to have.

**Comment by engelgabriel** [comment]
We can do a shell upgrade script inside our Docker Images, and trigger them from inside the UI like Discourse.

But the best way to update Docker Containers would still be by using Docker to download new images.

**Comment by bott0r** [comment]
That what i have been doing, i just have a shell script and a cronjob every tuesday morning.

**Comment by brylie** [comment]
Would you mind sharing your shell script example @bott0r?

**Comment by engelgabriel** [comment]
Please share?

**Comment by IQ2022** [comment]
I ran into an issue today with Heroku chat upgrade to new Release, and with one click was able to roll back from the admin interface, also was able to download mongodb DB - backup.. separate UI instance to manage the build would be really good..

**Comment by purldator** [comment]
@brylie & @engelgabriel: I use the following two commands to update my rocketchat container to latest after I `cd /var/www/rocket.chat`.  

````
docker pull rocketchat/rocket.chat:latest
docker-compose up --no-deps -d rocketchat
````

If one followed [the docker install instructions](https://rocket.chat/docs/installation/docker-containers/), then the above should work (as it do for mine).
`--no-deps` makes sure the mongodb container and hubot container are not touched; only rocketchat.

Throw that into an `.sh` file, have `#!/usr/bin/env bash` at the top, possibly `-p nameofyourproject` before `up`, save and run whenever or via cron job.

**Comment by TwizzyDizzy** [comment]
@rocket-cat close

First to everybody. If you still want to pursue this general topic of auto-updating, please check out #1581.

@brylie

I am closing this issue because we think there is no actual bug in Rocket.Chat. We understand, that setting up and operating a Rocket.Chat server is not trivial, yet we do not have the (wo)man-power to help everybody. That is unfortunate, but simply a limitation of time. If it should turn out from your feedback to this message, that there is a bug involved, we will try to fix it with you.

That being said, we are very thankful that our community also does a great job in helping each other. I would therefore like to point you to some places where you can try asking for help. Usually, some friendly human being has time for you and your issue:

* **The Rocket.Chat Community Support Channel**: https://open.rocket.chat/channel/support
* **The Rocket.Chat Community Support Forum**: https://forums.rocket.chat/c/community-support

If you don't find a solution to your problem there, don't fret, as we also offer paid support: https://rocket.chat/support

You could, of course, spare yourself the hassle entirely and just choose the hosted version, where we provide you with a running Rocket.Chat instance: https://rocket.chat/cloud

Cheers
Thomas

**Comment by brylie** [comment]
> I am closing this issue because we think there is no actual bug in Rocket.Chat.

This was a feature request.

**Comment by TwizzyDizzy** [comment]
Hi @brylie 

I know. I have been combing through issues the past days and have found lots of similar issues. In this case, I have closed the issue at hand in favour of #1581 (as I have mentioned), as it concerns itself with the same issue. The overall topic is: Auto-update of Rocket.Chat. This topic is preserved for future action in #1581. We cannot have multiple issues with the same topic.

In cases where there are already several issues with the same topic, it's sometimes hard to decide which issue to preserve and sometimes it may not be the right decision. Yet we need to cut down on the sheer number of issues and as long as the topic is preserved, I think it's fine.

What do you think? :)

Cheers
Thomas

**Comment by brylie** [comment]
Sounds good. Auto-update would be better than manual update, even if through an admin UI.

Cheers!
